src/app/debug/table/table.component.ts

```diff
@@ -215,10 +219,31 @@ export class TableComponent implements OnInit, OnDestroy {
       next: (settings) => {
         this.tableSettings.reportsInProgress = settings.reportsInProgress;
         this.tableSettings.estimatedMemoryUsage = settings.estMemory;
+        this.loadReportInProgressDates();
       },
     });
   }
 
+  loadReportInProgressDates() {
+    let hasChanged: boolean = false;
+    for (let i = 1; i <= this.tableSettings.reportsInProgress; i++) {
```

Comment:
Can't you rather use a for of loop?
I see that u need an index, but thats still possible.


```typescript
for (let [index, value] of myArray.entries()) {
  // your code goes here    
}
```

or 
```typescript
myArray.forEach((value, index) => {
  // your code goes here  
});
```

Reply:
Could you clarify the use of a for of/for each loop in this scenario? because the variable this.tableSettings.reportsInProgress is a number.

Reply:
When using a `for .. of` loop, each iteration of the iterable will be made available in a variable. You won't need to use `this.reportsInProgress[report.correlationId]` anymore. When using a destruct, you can also get an index.

Reply:
You told me in person that you are setting the values with this loop, and that the number updates separately.